### PR TITLE
Fixed (string) catchable fatal error for PHP Incomplete Class instances

### PR DESCRIPTION
**Pull request by yceruto**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #17586 |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

**Comment by yceruto** [comment]
AppVeyor build failed (not related)

**Comment by javiereguiluz** [comment]
:+1:

Status: reviewed

File: src/Symfony/Component/HttpKernel/DataCollector/Util/ValueExporter.php [hunk 1]
```diff
@@ -75,4 +79,11 @@ public function exportValue($value, $depth = 1, $deep = false)
 
         return (string) $value;
     }
+
+    private function getClassNameFromIncomplete(\__PHP_Incomplete_Class $value)
+    {
+        $array = new \ArrayObject($value);
+
+        return $array['__PHP_Incomplete_Class_Name'];
```
Comment (Tobion) [inline src/Symfony/Component/HttpKernel/DataCollector/Util/ValueExporter.php]: cant u just access this as a property without arrayobject?

Comment (javiereguiluz) [inline src/Symfony/Component/HttpKernel/DataCollector/Util/ValueExporter.php]:
```php
$foo = new \__PHP_Incomplete_Class();
echo $foo->__PHP_Incomplete_Class_Name;
```

results in:

```
The script tried to execute a method or access a property of an incomplete object.
Please ensure that the class definition "unknown" of the object you are trying to
operate on was loaded _before_ unserialize() gets called or provide a __autoload()
function to load the class definition.
```

Comment (stof) [inline src/Symfony/Component/HttpKernel/DataCollector/Util/ValueExporter.php]: yeah, __PHP_Incomplete_Class is really weird (the fact that ``is_object`` returns false for it is a first WTF, but not the only one)

**Comment by Tobion** [comment]
:+1:

**Comment by fabpot** [comment]
This one should be merged in 2.7, but first, we need a PR on 2.3 as the bug probably exists there as well.

**Comment by yceruto** [comment]
`ValueExporter.php` has been inserted from 2.4 version ([commit](https://github.com/symfony/symfony/commit/f56c5774a8ae0717c264f61af2ec0f9aa9f125f7))

**Comment by stof** [comment]
@yceruto but the bug also exists in 2.3. It is just not extracted to a separate class

**Comment by yceruto** [comment]
@stof right, I will create a PR to 2.3

**Comment by yceruto** [comment]
I opened a PR for 2.3 branch (https://github.com/symfony/symfony/pull/17766)

**Comment by fabpot** [comment]
Thank you @yceruto.

**Pull request closed without merging**